presentation/src/main/java/com/github/ianellis/shifts/presentation/ShiftsViewModel.kt

```diff
@@ -36,42 +40,42 @@ class ShiftsViewModel constructor(
     private var shiftUpdateJob: Job? = null
 
     init {
-        loadShifts()
-    }
+        shifts.value = emptyList()
+        shiftState.value = CurrentShiftState.OFF
 
-    private val onShiftsChanged: (Event<List<Shift>>) -> Unit = { event ->
         launch {
-            event.value?.let {
-                shifts.value = it
-                if (shifts.value?.lastOrNull()?.state is ShiftState.Started) {
-                    shiftState.value = CurrentShiftState.STARTED
+            loadShifts()
+        }
+
+        launch {
+            shiftsUpdated.consumeEach { event ->
+
+                event.value?.let {
+                    shifts.value = it
+                    if (shifts.value?.lastOrNull()?.state is ShiftState.Started) {
+                        shiftState.value = CurrentShiftState.STARTED
+                    }
                 }
-            }
 
-            event.error?.let {
-                showGenericError.value = event.error
-            }
+                event.error?.let {
+                    showGenericError.value = event.error
+                }
 
-            loading.value = false
+                loading.value = false
+            }
         }
-    }
 
-    private val ignoreShifts: Ignore
-
-    init {
-        shifts.value = emptyList()
-        shiftState.value = CurrentShiftState.OFF
-        ignoreShifts = listenForShifts(onShiftsChanged)
         loading.value = true
     }
 
     fun startShift(v: View? = null) {
```

Comment:
ViewModel should not have references to views.

Reply:
Good point, the reason I did this is so you can bind to the viewModel using a method reference, in the xml I can then bind the click handler like so "android:onClick="@{viewModel::startShift"}

Otherwise, you need to roll your own binding, which can potentially cause leaks across configuration changes

Reply:
I did a bit more research and found you can use lambda syntax is the xml to avoid the view param in the view model. I now have it as such ` android:onClick="@{()->viewModel.startShift()}"